src/core/context/AppContextProvider.tsx

```diff
@@ -194,13 +231,8 @@ export const AppContextProvider = (props: {
   }, []);
 
   useEffect(() => {
-    if (!messageHistoryStore.getSnapshot().length) {
-      messageHistoryStore.addMessage({
-        role: 'system' as const,
-        content: modelConfig.systemPrompt,
-      });
-    }
-  }, [messageHistoryStore, modelConfig.systemPrompt]);
+    ensureCorrectSystemPrompt(messageHistoryStore, modelConfig);
+  }, [messageHistoryStore, modelConfig, ensureCorrectSystemPrompt]);
```

Comment:
I can't decide if this useEffect is needed anymore. The only times we should be resetting and adding the systemPrompt in is on new chat started and on model change which are both covered above. Makes me think this is a redundant update. Thoughts?

Reply:
Before seeing this comment, I was thinking "Do we still need this?"

I did some manual testing with it removed and it seems good.